mne/event.py

```diff
@@ -814,8 +814,12 @@ def shift_time_events(events, ids, tshift, sfreq):
         The new events.
     """
     events = events.copy()
-    for ii in ids:
-        events[events[:, 2] == ii, 0] += int(tshift * sfreq)
+    if ids is None:
+        events[:, 0] += int(tshift * sfreq)
+    elif isinstance(ids, list):
+        for ii in ids:
+            events[events[:, 2] == ii, 0] += int(tshift * sfreq)
```

Comment:
A nice DRY change would be to change the above branch mask=slice(None) and this one mask=np.in1d(events[:,2],IDs), then do the += below.

Reply:
(on mobile so please forgive the errors, hopefully you get the idea)